tests/solidity/test_close_short.py

```diff
@@ -0,0 +1,364 @@
+        This function is run before each test method.
+        """
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        block_time = time.BlockTime()
+        pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.0,
+            redemption_fee_percent=0.0,
+        )
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=pricing_model,
+            market_state=market_state,
+            position_duration=time.StretchedTime(
+                days=365, time_stretch=pricing_model.calc_time_stretch(self.target_apr), normalizing_constant=365
+            ),
+            block_time=block_time,
+        )
+        _, wallet_deltas = self.hyperdrive.initialize(self.alice.wallet.address, self.contribution, 0.05)
+        self.alice.wallet.update(wallet_deltas)
+
+    def verify_close_short(
+        self,
+        example_agent: agent.Agent,
+        market_state_before: hyperdrive_market.MarketState,
+        agent_base_paid: float,
+        agent_base_proceeds: float,
+        bond_amount: float,
+        maturity_time: float,
+    ):
+        """Close a short then make sure the market state is correct"""
+        # verify that all of Bob's bonds were burned
+        self.assertFalse(
+            example_agent.wallet.shorts
+        )  # In solidity we check that the balance is zero, but here we delete the entry if it is zero
+        # verify that the bond reserves were updated according to flat+curve
+        # the adjustment should be equal to timeRemaining * bondAmount
+        time_remaining = 0
+        if maturity_time > self.hyperdrive.block_time.time:
+            time_remaining = self.hyperdrive.position_duration.normalized_time * (
+                maturity_time - self.hyperdrive.block_time.time
+            )
+        self.assertLess(  # user gets less than what they put in
+            agent_base_proceeds,
+            agent_base_paid,
+            msg="agent gets more than what they put in: agent_base_proceeds > agent_base_paid",
+        )
+        self.assertEqual(  # bond reserves
+            self.hyperdrive.market_state.bond_reserves,
+            market_state_before.bond_reserves - time_remaining * bond_amount,
+            msg="bond_reserves is wrong",
+        )
+        self.assertEqual(  # share reserves
+            self.hyperdrive.market_state.share_reserves,
+            market_state_before.share_reserves + (bond_amount - agent_base_proceeds) / market_state_before.share_price,
+            msg="share_reserves is wrong",
+        )
+        self.assertEqual(  # lp total supply
+            self.hyperdrive.market_state.lp_total_supply,
+            market_state_before.lp_total_supply,
+            msg="lp_total_supply is wrong",
+        )
+        self.assertEqual(  # share price
```

Comment:
I don't think this assert should be in here. We allow the share price to change when the market advances. It's also not in solidity

Reply:
oh yeah this was removed when they added negative interest tests